_posts/2019-10-16-deployment-updates.md

```diff
@@ -0,0 +1,225 @@
+SGE, PBS, SLURM, LSF, some cloud service or some custom system.
+
+As Dask is used by more institutions and used more broadly within those
+institutions making deployment smooth and natural becomes increasingly
+important.  This is so important in fact, that there have been seven separate
+efforts to improve deployment in some regard or another by a few different
+groups.
+
+We'll briefly summarize and link to this work below, and then we'll finish up
+by talking about some internal design that helped to make this work more
+consistent.
+
+
+Dask-SSH
+--------
+
+According to our user survey, the most common deployment mechanism was still
+SSH.  Dask has had a [command line dask-ssh
+tool](https://docs.dask.org/en/latest/setup/ssh.html#command-line) to make it
+easier to deploy with SSH for some time.  We recently updated this to also
+include a Python interface, which provides more control.
+
+```python
+>>> from dask.distributed import Client, SSHCluster
+>>> cluster = SSHCluster(
+...     ["host1", "host2", "host3", "host4"],
+...     connect_options={"known_hosts": None},
+...     worker_options={"nthreads": 2},
+...     scheduler_options={"port": 0, "dashboard_address": ":8797"}
+... )
+>>> client = Client(cluster)
+```
+
+This isn't what we recommend for large institutions, but it can be helpful for
+more informal groups who are just getting started.
+
+
+Dask-Jobqueue and Dask-Kubernetes Rewrite
+-----------------------------------------
+
+We've rewritten Dask-Jobqueue for SLURM/PBS/LSF/SGE cluster managers typically
+found in HPC centers and Dask-Kubernetes.  These now share a common codebase
+along with Dask SSH, and so are much more consistent and hopefully bug free.
+
+Ideally users shouldn't notice much difference with existing workloads,
+but new features like asynchronous operation, integration with the Dask
+JupyterLab extension, and so on are more consistently available.  Also, we've
+been able to unify development and reduce our maintenance burden considerably.
+
+The new version of Dask Jobqueue where these changes take place is 0.7.0, and
+the work was done in [dask/dask-jobqueue #307](https://github.com/dask/dask-jobqueue/pull/307).
+The new version of Dask Kubernetes is 0.10.0 and the work was done in
+[dask/dask-kubernetes #162](https://github.com/dask/dask-kubernetes/pull/162).
+
+
+Dask-CloudProvider
+------------------
+
+For cloud deployments we generally recommend using a hosted Kubernetes or Yarn
+service, and then using Dask-Kubernetes or Dask-Yarn on top of these.
+
+However in some institutions they have made decisions or commitments to use
+certain vendor specific technologies, and it's more convenient to use APIs that
+are more native to the particular cloud.  The new package [Dask
+Cloudprovider](https://cloudprovider.dask.org) handles this today for Amazon's
+ECS API, which has been around for a long while and is more universally
+accepted.
```

Comment:
Maybe that's not to be put here, but does it handles also adaptivity and easy scaling?

Reply:
It is based on SpecCluster, so programmatically yes.  However I suspect that the time to create new workers varies depending on the system used.  One of the systems, Fargate, is intended to be fairly responsive I believe.